programs/openbook-v2/src/instructions/settle_funds.rs

```diff
@@ -7,6 +7,21 @@ use crate::token_utils::*;
 
 pub fn settle_funds<'info>(ctx: Context<'_, '_, '_, 'info, SettleFunds<'info>>) -> Result<()> {
     let mut open_orders_account = ctx.accounts.open_orders_account.load_mut()?;
+
+    // ensure delegate can only withdraw to owner's token accounts
```

Comment:
I think this can be implemented in a similar was as `is_owner_or_delegate` and so it can be called from the accounts_ix. So all these checks are done in those other files.